tornado/ioloop.py

```diff
@@ -1162,24 +1201,30 @@ class _Timeout(object):
     """An IOLoop timeout, a UNIX timestamp and a callback"""
 
     # Reduce memory overhead when there are lots of pending callbacks
-    __slots__ = ['deadline', 'callback', 'tdeadline']
+    __slots__ = ['deadline', 'callback', 'seqno']
 
     def __init__(self, deadline, callback, io_loop):
-        if not isinstance(deadline, numbers.Real):
+        if not isinstance(deadline, Real):
```

Comment:
On line 646, you check for (float, Real). Why do you only check for real here?

Reply:
Line 646 is an optimization. I guess I could do the same here.